src/service/inputs/parsed-inputs.ts

```diff
@@ -0,0 +1,25 @@
+import { defaultInputValues, InputValues, LoggerLevel } from "@bc/domain/inputs";
+import { Service } from "typedi";
+
+@Service()
+export class ParsedInputs {
+    // store parsed options
+    private _inputs: InputValues = defaultInputValues;
+
+    /**
+     * Getter for parsed options
+     * @returns Object containing parsed options
+     */
+    get inputs(): InputValues {
+        return this._inputs;
+    }
+
+    /**
+     * Updates the parsed options
+     * @param updatedInputs Options that were obtained from the parser
+     */
+    updateInputs(updatedInputs: InputValues) {
+        this._inputs = {...this.inputs, ...updatedInputs};
+        if (updatedInputs.debug) {this._inputs = {...this._inputs, loggerLevel: LoggerLevel.DEBUG};}
```

Comment:
isn't it already done at `action-arguments.ts#L84`? in any case this should loggerLevel agnostic, this class should be in charge of handling inputs and only that, shouldn't know anything about loggerLevel at all. Don't you think so?

Reply:
action-arguments is used to get the input for during github action and according to action.yml doesn't really use `debug` as an option. This piece of code is mainly for CLI which has a debug flag. So instead of having 2 different keys to look at (debug for CLI and loggerLevel for action), when CLI debug is parsed it sets the logger level to debug. This way we always just have to look at the loggerLevel key only.

I would also argue that this should be part of the input service since it is parsing the debug flag for the CLI. The CLI argument parser should only provide the user values for all the flags after which it is upto the input service on how to perfectly set those values such that they can be used by the configuration reader later on